flytekit/types/file/file.py

```diff
@@ -435,6 +435,8 @@ def to_literal(
                     f" specified. Since a simpler type was specified, we'll skip uploading!"
                 )
                 should_upload = False
+            if ctx.execution_state.is_local_execution and python_val.remote_path is None:
+                should_upload = False
```

Comment:
Should we set should_upload to `False` by default instead? That way, we won't need `should_upload=False` everywhere.